node/service/src/service.rs

```diff
@@ -262,9 +266,39 @@ async fn start_node_impl(
 		})
 		.await?;
 
+	// Start off-chain workers if enabled
+	if parachain_config.offchain_worker.enabled {
+		log::info!("OFFCHAIN WORKER is Enabled!");
+		let offchain_workers =
+			sc_offchain::OffchainWorkers::new(sc_offchain::OffchainWorkerOptions {
+				runtime_api_provider: client.clone(),
+				is_validator: validator,
+				keystore: Some(params.keystore_container.keystore()),
+				offchain_db: backend.offchain_storage(),
+				transaction_pool: Some(OffchainTransactionPoolFactory::new(
+					transaction_pool.clone(),
+				)),
+				network_provider: network.clone(),
+				enable_http_requests: true,
+				custom_extensions: |_| vec![],
+			});
+
+		// Spawn a task to handle off-chain notifications.
+		// This task is responsible for processing off-chain events or data for the blockchain.
+		task_manager.spawn_handle().spawn(
+			"offchain-workers-runner",
+			"offchain-work",
+			offchain_workers.run(client.clone(), task_manager.spawn_handle()).boxed(),
```

Comment:
Can this throw an error?  If so, can it be caught/logged?

`    log::error!("Error running off-chain workers: {}", err);
`

Reply:
based on what I can see spawn gets a task of `Future<Output = ()> ` type. So it does not return a Result type and the inner code provided by substrate doesn't seem to have a panic scenario that can be handled from anywhere inside here.

Reply:
yeah spawn will create a tokio task and run it until it ends , i think task_manager sent to offchain_worker manages that, just a thought